pkg/lightning/restore/restore.go

```diff
@@ -2636,6 +2637,17 @@ func (cr *chunkRestore) encodeLoop(
 				return
 			}
 			kvPacket = append(kvPacket, deliveredKVs{kvs: kvs, columns: columnNames, offset: newOffset, rowID: rowID})
+			kvSize += kvs.Size()
+			failpoint.Inject("mock-kv-size", func(val failpoint.Value) {
+				kvSize += uint64(val.(int))
+			})
+			// pebble cannot allow > 4.0G kv in one batch.
+			// we will meet pebble panic when import sql file and each kv has the size larger than 4G / maxKvPairsCnt.
+			// so add this check.
+			if kvSize > minDeliverBytes {
```

Comment:
```suggestion
			if kvSize >= minDeliverBytes || len(kvPacket) >= maxKvPairsCnt || newOffset == cr.chunk.Chunk.EndOffset {
				canDeliver = true
				kvSize = 0
			}
```

Reply:
Please merge the two if together

Reply:
kvPacket contains both `dataKVs` and `indexKVs` here. Since `dataKVs` and `indexKVs` are wrote to different engines, how about moving this check to `deliverLoop`?